web-app/js/contacts.js

```diff
@@ -360,7 +360,10 @@ $(function () {
         var $button = $(this);
         var linkId = $button.attr("data-linkId");
         var $link = $("#"+linkId);
-        resend($link, $button);
+        var resendStatus = resend($link, $button);
+        if(resendStatus) {
+            $("#activity-resend-confirm").modal("hide");
+        }
```

Comment:
This code doesn't work as you expect. 
Resend is an asyncronous method. So resend always returns true. The function .done() can't return anything because it si async. It needs to receibe a callback function that will be exectuted on success or on fail.